frontend/public/components/nav/PluginNavItem.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { NavItemSeparator } from '@patternfly/react-core';
+import {
+  isNavSection,
+  isSeparator,
+  NavExtension,
+  isHrefNavItem,
+  isResourceNSNavItem,
+  isResourceNavItem,
+} from '@console/dynamic-plugin-sdk';
+import { LoadedExtension } from '@console/dynamic-plugin-sdk/src/types';
+import { NavItemHref } from './NavItemHref';
+import { NavSection } from './NavSection';
+import { NavItemResource } from './NavItemResource';
+
+export const PluginNavItem: React.FC<PluginNavItemProps> = ({ extension }) => {
+  if (isNavSection(extension)) {
+    return (
+      <NavSection
+        id={extension.properties.id}
+        name={extension.properties.name}
+        dataAttributes={extension.properties.dataAttributes}
+      />
+    );
+  }
+  if (isSeparator(extension)) {
+    return <NavItemSeparator key={extension.uid} {...extension.properties.dataAttributes} />;
+  }
+  if (isHrefNavItem(extension)) {
+    return (
+      <NavItemHref
+        href={extension.properties.href}
+        namespaced={extension.properties.namespaced}
+        prefixNamespaced={extension.properties.prefixNamespaced}
+        startsWith={extension.properties.startsWith}
+        dataAttributes={extension.properties.dataAttributes}
+      >
+        {extension.properties.name}
+      </NavItemHref>
+    );
+  }
+  if (isResourceNavItem(extension)) {
+    return (
+      <NavItemResource
+        namespaced={isResourceNSNavItem(extension)}
+        model={extension.properties.model}
+        startsWith={extension.properties.startsWith}
+        dataAttributes={extension.properties.dataAttributes}
+      >
+        {extension.properties.name}
+      </NavItemResource>
+    );
+  }
+  return null;
```

Comment:
Curious, if it makes it here, is this a valid state or did we want to also `console.warn` or something that we didn't render an unknown nav item?

Reply:
Since this is an used component and we use type guards to make sure we are only passing nav extensions to this component, we probably don't need a warning here. We theoretically shouldn't ever reach this line.

Reply:
More of a reason I was thinking of a warning -- React does warnings for components in the render thread by having a flag they switch when it's been presented once. I was thinking we could do that.

But we can ignore this idea for now. Less of a jump on any bug that is related in the future, but probably not by much.